### ISSUE TITLE
Update Asio/Boost version

### ISSUE THREAD

**Issue by prince-chrismc** [Stiffstream/restinio-conan#13]
Currently RESTinio is capabled of running on asio/1.16 or boost/1.72 but the version specified in the requirements are slightly dated.

**Comment by eao197** [comment]
I don't use Conan for myself so I don't know is it possible to override dependencies in user's conanfile.txt? Something like:
```
[requires]
restinio/0.6.4@stiffstream/stable
boost/1.72.0
```

**Comment by prince-chrismc** [comment]
That is correct but it would be nice to not manually manage the dependencies of my dependencies. It's the reason I opted for Conan.

This is a nice to have enhancement.

**Comment by eao197** [comment]
I'm afraid of that case: someone finds RESTinio and wants to use it with Boost.Asio. But has to stay at some older version of Boost. If we specify the latest version of Boost in RESTinio's recipe then such user can reject RESTinio.

So I more like the idea of specifying the oldest of the supported versions of Boost for RESTinio. If someone wants to use a newer version he or she can override the version in local conanfile.txt.

But I open to a discussion.

PS. I don't remember why RESTinio uses Boost-1.69 as a dependency instead of 1.66. Becase 1.66 is the oldest supported version of Boost.

**Comment by prince-chrismc** [comment]
That's a very suitable argument.

Perhaps the solution is to pick either the highest or the lowest and document the ability to select the asio and boost versions? Currently the only way to figure that details out is from the release note of the main RESTinio repo. [this is the only place it's written](https://github.com/Stiffstream/restinio/blob/8c9df128e4c9d37b42ae939f1c9937c93ffe8f20/dev/restinio/boost_helper.rb#L23) for boost

**Comment by prince-chrismc** [comment]
Not that I have seen this very often but a possibility would be [version ranges](https://docs.conan.io/en/latest/versioning/version_ranges.html)

**Comment by Croydon** [comment]
Please don't use version ranges
I think that it should be a goal to get this Conan recipe into the Conan Center Index and it doesn't allow version ranges for a variety of reasons

**Comment by prince-chrismc** [comment]
@Croydon for Conan Center Index, is there a rule or guideline for dependency version? Following that convention would make moving this recipe easier.

**Comment by Croydon** [comment]
I guess it can be summarizes as this:
  * no system_requirements()
  * Conan dependencies need to be in the Conan Center Index too (not from other Conan repositories)
  * no version ranges
  * no arbitrary dependencies e.g. set via options at runtime

I theory I guess you could specify several versions of a dependency as an option
```
  options = {
    use_boost_version = ["1.69.0", "1.70.0"]
    ...
  default_options = {
     use_boost_version = "1.70.0"
    ....

   requires(self):
     self.requires.add("boost/{}").format(self.options.use_boost_version) 
```

But I think that the advantages of this are very little and I would not recommend this.
If people really need it they can override it via their conanfile.
I believe there are some recipes in other repositories doing something like this, but I don't recall that we have one in CCI which is doing this right now.

**Comment by prince-chrismc** [comment]
What Stiffstream is proposing then is acceptable, however the lowest boost available is 1.69.0 from CCI. That would imply no changes to the recipe. Yay!

**Comment by prince-chrismc** [comment]
fixed 18490a28c0c5f191995f830a85a117ccfd5d3d35
